lib/streamlit/runtime/fragment.py

```diff
@@ -105,9 +108,73 @@ def fragment(
 
 
 @gather_metrics("experimental_fragment")
-def fragment(  # type: ignore[empty-body]
+def fragment(
     func: F | None = None,
     *,
     run_every: float | None = None,
 ) -> Callable[[F], F] | F:
-    pass
+    """TODO(vdonato): Write a docstring for this function."""
+
+    if func is None:
+        # Support passing the params via function decorator
+        def wrapper(f: F) -> F:
+            return fragment(
+                func=f,
+                run_every=run_every,
+            )
+
+        return wrapper
+    else:
+        non_optional_func = func
+
+    @wraps(non_optional_func)
+    def wrap(*args, **kwargs):
```

Comment:
I'm not 100% sure if it is really needed, but it might be that you need to add something like this to preserve certain function attributes for the wrapped function:

https://github.com/streamlit/streamlit/blob/5e6830f20241b8ab4ff4510f7c6e12644b1926a9/lib/streamlit/runtime/metrics_util.py#L415

Reply:
Hm, yeah I'm also not entirely sure if this is strictly required, but I doubt it'll hurt to preserve the original function's signature / `__dict__` attributes. Will add these lines